components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/GatewayStartupListener.java

```diff
@@ -99,4 +98,38 @@ public void invoke() {
         }
     }
 
-}
+
+    public void startListener() {
+        new Thread(this).start();
+    }
+
+    @Override
+    public void run() {
+        deployArtifactsInGateway();
+    }
+
+    private void deployArtifactsInGateway() {
+        long retryDuration = 10000;
```

Comment:
can we bring it to config level

Reply:
https://github.com/wso2/carbon-apimgt/pull/8828/commits/3540ddaf35c95330c5411a866aaeb384dad47ee3